e2e_playwright/st_download_button_test.py

```diff
@@ -66,31 +66,39 @@ def test_value_not_reset_on_reclick(app: Page):
     expect(app.get_by_test_id("stMarkdown").first).to_have_text("value: True")
 
 
-def test_click_calls_callback(app: Page):
+def test_value_correct_on_ignore_click(app: Page):
     download_button = app.get_by_test_id("stDownloadButton").nth(12).locator("button")
-    expect(app.get_by_test_id("stMarkdown").nth(3)).to_contain_text(
+    download_button.click()
+    expect(app.get_by_test_id("stMarkdown").nth(1)).to_have_text(
+        "Ignore rerun download button value: False"
+    )
+
+
+def test_click_calls_callback(app: Page):
+    download_button = app.get_by_test_id("stDownloadButton").nth(13).locator("button")
```

Comment:
nit: there is a `get_element_by_key` utility function now in our `app_utils`. I think it might be good for new tests to leverage that function instead of continuing to use `nth(...)` since it makes it easier to maintain and reason about the tests

Reply:
done